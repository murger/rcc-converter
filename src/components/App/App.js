import React, { Component, Fragment } from 'react'
import { ThemeProvider } from 'styled-components'
import { GlobalStyles, theme } from '../../theme'
import { ServiceProvider } from '../../contexts/ServiceContext'
import Converter from '../Converter'

class App extends Component {
  render () {
    return (
      <ThemeProvider theme={theme}>
        <Fragment>
          <GlobalStyles />
          <ServiceProvider>
            <Converter />
          </ServiceProvider>
        </Fragment>
      </ThemeProvider>
    )
  }
}

export default App
