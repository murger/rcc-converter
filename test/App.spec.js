import React from 'react'
import renderer from 'react-test-renderer'
import App from '../src/components/App'

// react-text-mask has the best UX but it doesn't yield well into testing
// https://github.com/text-mask/text-mask/issues/427
jest.mock('react-text-mask', () => props => <input {...props } />)

// Snapshot tests are good for catching unintended changes
it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
