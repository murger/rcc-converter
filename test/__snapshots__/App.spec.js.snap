// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="sc-bdVaJa gnNedY"
  >
    <div
      className="sc-bwzfXH kqxhZx"
    >
      <p
        className="sc-htpNat brQbym"
        onClick={[Function]}
      >
        GBP
      </p>
      <p
        className="sc-htpNat fwGNFT"
        onClick={[Function]}
      >
        USD
      </p>
      <p
        className="sc-htpNat fwGNFT"
        onClick={[Function]}
      >
        EUR
      </p>
      <p
        className="sc-htpNat fwGNFT"
        onClick={[Function]}
      >
        JPY
      </p>
    </div>
    <div
      className="sc-bwzfXH bLDufQ"
      width="70%"
    >
      <div
        className="sc-EHOje fRMCnM"
      >
        Press ↑↓ to change currency or ↵ to buy 
        <b>
          ¥
        </b>
         @ 
        0.00000
      </div>
      <input
        autoFocus={true}
        className="sc-ifAKCX cLThwN"
        mask={[Function]}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        value={null}
      />
      <p
        className="sc-bxivhb kSLlTs"
      >
        You have 
        £5,000.00
      </p>
    </div>
  </div>,
  <div
    className="sc-bdVaJa lmiDtV"
  >
    <div
      className="sc-bwzfXH kqxhZx"
    >
      <p
        className="sc-htpNat fwGNFT"
        onClick={[Function]}
      >
        GBP
      </p>
      <p
        className="sc-htpNat fwGNFT"
        onClick={[Function]}
      >
        USD
      </p>
      <p
        className="sc-htpNat fwGNFT"
        onClick={[Function]}
      >
        EUR
      </p>
      <p
        className="sc-htpNat brQbym"
        onClick={[Function]}
      >
        JPY
      </p>
    </div>
    <div
      className="sc-bwzfXH bLDufQ"
      width="70%"
    >
      <div
        className="sc-EHOje fRMCnM"
      >
        Press ↑↓ to change currency or ↵ to buy 
        <b>
          £
        </b>
         @ 
        0.00000
      </div>
      <input
        autoFocus={false}
        className="sc-ifAKCX cLThwN"
        mask={[Function]}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        value={null}
      />
      <p
        className="sc-bxivhb kSLlTs"
      >
        You have 
        ¥0
      </p>
    </div>
  </div>,
]
`;
